instrumentation/apache-httpclient/apache-httpclient-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachehttpclient/v2_0/ApacheHttpClientNetAttributesExtractor.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.apachehttpclient.v2_0;
+
+import io.opentelemetry.instrumentation.api.instrumenter.net.NetAttributesExtractor;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import org.apache.commons.httpclient.HostConfiguration;
+import org.apache.commons.httpclient.HttpMethod;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+final class ApacheHttpClientNetAttributesExtractor
+    extends NetAttributesExtractor<HttpMethod, Void> {
+
+  @Override
+  protected String transport(HttpMethod httpMethod) {
+    return SemanticAttributes.NetTransportValues.IP_TCP;
+  }
+
+  @Override
+  protected @Nullable String peerName(HttpMethod httpMethod, @Nullable Void unused) {
+    HostConfiguration hostConfiguration = httpMethod.getHostConfiguration();
+    return hostConfiguration != null ? hostConfiguration.getHost() : null;
+  }
+
+  @Override
+  protected @Nullable Long peerPort(HttpMethod httpMethod, @Nullable Void unused) {
+    HostConfiguration hostConfiguration = httpMethod.getHostConfiguration();
+    return hostConfiguration != null ? (long) hostConfiguration.getPort() : null;
+  }
+
+  @Override
+  protected @Nullable String peerIp(HttpMethod httpMethod, @Nullable Void unused) {
+    return null;
+  }
```

Comment:
Should we detect if `host` is an IP address and use `net.peer.ip` instead of `net.peer.name` in that case?

Reply:
This has always been tricky - I think if the framework doesn't provide us a resolved address (e.g., `SocketAddress`) then we'd just populate `net.peer.name`, even if it's an IP address. Not sure if that's great though.